Simplified/NYPLNetworkQueue.swift

```diff
@@ -0,0 +1,173 @@
+  private static let sqlHeader = Expression<Data?>("requestHeader")
+  private static let sqlRetries = Expression<Int>("retryCount")
+  
+  
+  // MARK: - Public Functions
+  
+  class func addRequest(_ libraryID: Int,
+                        _ updateID: String?,
+                        _ requestUrl: URL,
+                        _ method: HTTPMethodType,
+                        _ parameters: Data?,
+                        _ headers: [String : String]?) -> Void
+  {
+    // Serialize Data
+    let urlString = requestUrl.absoluteString
+    let methodString = method.rawValue
+    
+    let headerData: Data?
+    if headers != nil {
+      headerData = NSKeyedArchiver.archivedData(withRootObject: headers!)
+    } else {
+      headerData = nil
+    }
+    
+    guard let db = startDatabaseConnection() else { return }
+    
+    // Get or create table
+    do {
+      try db.run(sqlTable.create(ifNotExists: true) { t in
+        t.column(sqlID, primaryKey: true)
+        t.column(sqlLibraryID)
+        t.column(sqlUpdateID)
+        t.column(sqlUrl)
+        t.column(sqlMethod)
+        t.column(sqlParameters)
+        t.column(sqlHeader)
+        t.column(sqlRetries)
+      })
+    } catch {
+      Log.error(#file, "SQLite Error: Could not create table")
+      return
+    }
+    
+    // Update (not insert) if uniqueID and libraryID match existing row in table
+    let query = sqlTable.filter(sqlLibraryID == libraryID && sqlUpdateID == updateID)
+                        .filter(sqlUpdateID != nil)
+    
+    do {
+      // Update row
+      if try db.run(query.update(sqlParameters <- parameters, sqlHeader <- headerData)) > 0 {
+        Log.info(#file, "SQLite Row Updated - Success")
+        
+      // Insert new row
+      } else {
+        do {
+          try db.run(sqlTable.insert(sqlLibraryID <- libraryID, sqlUpdateID <- updateID, sqlUrl <- urlString, sqlMethod <- methodString, sqlParameters <- parameters, sqlHeader <- headerData, sqlRetries <- 0))
+          Log.info(#file, "SQLite Row Added - Success")
+        } catch {
+          Log.error(#file, "SQLite Error: Could not update table")
+        }
+      }
+    } catch {
+      Log.error(#file, "SQLite Error: Could not update queue")
+    }
+  }
+  
+  class func retryQueue()
+  {
+    guard let db = startDatabaseConnection() else { return }
+    
+    do {
+      for row in try db.prepare(sqlTable) {
+        self.retry(db: db, requestRow: row)
+      }
+    } catch {
+      Log.error(#file, "SQLite Error accessing table")
+    }
+  }
+  
+  
+  // MARK: - Private Functions
+  
+  private class func retry(db: Connection, requestRow: Row)
+  {
+    if (Int(requestRow[sqlRetries]) > QueueMaxRetryCount) {
+      deleteRow(db: db, id: Int(requestRow[sqlID]))
+      Log.info(#file, "Removing after \(Int(requestRow[sqlRetries])) retries")
+      return
+    }
+    
+    do {
+      let ID = Int(requestRow[sqlID])
+      let newValue = Int(requestRow[sqlRetries]) + 1
```

Comment:
I think i am ok with the reties. if a users device has been off for a month and that device has un-posted reading positions, i would think that user won't be able to read that book anyways, that book might have expired already, and for numerous other reasons we should be good with the retries, and in case of there is a server outage for an hour or so, as far as i understood, that is not a reason to increase the retry, (correct me if i am wrong)